include/BSMPT/bounce_solution/action_calculation.h

```diff
@@ -0,0 +1,821 @@
+#define INCLUDE_BSMPT_GravitationalWaves_GWUtils_ActionCalculation_H_
+
+/**
+ * @brief This classes calculates the Bounce action of the potential with a set
+ * temperature
+ */
+
+namespace BSMPT
+{
+
+class BounceActionInt
+{
+private:
+  // These two variables check if the method undershot and overshot at least
+  // once. If only one type of convergence was achieved it is an indication that
+  // no solution has been found.
+  /**
+   * @brief Records if overshoot/undershoot method undershots at least once
+   */
+  bool undershot_once = false;
+  /**
+   * @brief Records if overshoot/undershoot method overshots at least once
+   */
+  bool overshot_once = false;
+
+  /**
+   * @brief Value of d2Vdl^2 near the true vacuum
+   *
+   */
+  double TrueVacuumHessian;
+
+  /**
+   * @brief Store the value of backwards propagation
+   *
+   */
+  double Initial_lmin;
+
+  /**
+   * @brief Potential at the false vacuum in the unshifted potential (in the
+   * shifted Vfalse = 0)
+   *
+   */
+  double Vfalse;
+
+  /**
+   * @brief l0 - Initial_lmin for solutions starting very near the true vacuum
+   *
+   */
+  double l0_Initial_lmin;
+
+  /**
+   * @brief True if path deformation reached the desired results without solving
+   * the 1D equation one more time.
+   *
+   */
+  bool PathDeformationConvergedWithout1D = false;
+
+public:
+  /**
+   * @brief Dimension of the VEV space
+   *
+   */
+  int dim = -1;
+
+  /**
+   * @brief Action calculated by the class
+   *
+   * * = -1 if not yet calculated.
+   * * = -2 if 1D integration did not work.
+   * * = -3 if the path deformation did not converged in time.
+   * * = -4 if the path deformation crashed.
+   * * = -5 if the false vacuum in not a minimum.
+   * * = -6 path deformation created a path with V(True) > V(False) (even with
+   * backwards propagation).
+   * * = -7 1D integration never overshot/undershot
+   * * = -8 overshooting/undershooting found a negative gradient point
+   * * = -9 if the solution does not have enough points to populate the \f$
+   * l(\rho) \f$ spline
+   */
+  double action = -1;
+
+  /**
+   * @brief Factor produced by the spherical symmetry of the potential.
+   * * = 2 if \f$ T > 0\f$ (\f$O(3)\f$ symmetry).
+   * * = 3 if \f$ T = 0\f$ (\f$O(4)\f$ symmetry).
+   *
+   * Default value is \f$ 2 \f$ since most of our calculation are done at finite
+   * temperature.
+   */
+  double alpha = 2; // alpha = 2 if T > 0 | alpha = 3 if T = 0
+
+  /**
+   * @brief Temperature of the potential. Irrelevant but helpful
+   *
+   */
+  double T = -1;
+
+  /**
+   * @brief Number of integration of the bounce
+   *
+   */
+  int MaxPathIntegrations;
+
+  /**
+   * @brief Number of path deformations before integrating again
+   *
+   */
+  int MaxSinglePathDeformations = 200;
+
+  /**
+   * @brief Do you want to save the solution path in a folder?
+   * * Yes = 1
+   * * No = 0
+   * (Currently not working)
+   *
+   */
+  int SaveOutput = 0;
```

Comment:
If it is currently not working, why keep it?
Also this looks like it should be an enum or a bool

Reply:
Removed.